internal/services/cognitive/cognitive_account_project_resource.go

```diff
@@ -0,0 +1,262 @@
+package cognitive
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/identity"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/location"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/cognitive/2025-06-01/cognitiveservicesaccounts"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/cognitive/2025-06-01/cognitiveservicesprojects"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/locks"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+type CognitiveAccountProjectModel struct {
+	Name               string                                     `tfschema:"name"`
+	CognitiveAccountId string                                     `tfschema:"cognitive_account_id"`
+	Location           string                                     `tfschema:"location"`
+	Identity           []identity.ModelSystemAssignedUserAssigned `tfschema:"identity"`
+	Tags               map[string]string                          `tfschema:"tags"`
+
+	Endpoints map[string]string `tfschema:"endpoints"`
+	IsDefault bool              `tfschema:"is_default"`
+}
+
+type CognitiveAccountProjectResource struct{}
+
+var _ sdk.ResourceWithUpdate = CognitiveAccountProjectResource{}
+
+func (r CognitiveAccountProjectResource) ResourceType() string {
+	return "azurerm_cognitive_account_project"
+}
+
+func (r CognitiveAccountProjectResource) ModelObject() interface{} {
+	return &CognitiveAccountProjectModel{}
+}
+
+func (r CognitiveAccountProjectResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return cognitiveservicesprojects.ValidateProjectID
+}
+
+func (r CognitiveAccountProjectResource) Arguments() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"name": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+			ValidateFunc: validation.StringMatch(
+				regexp.MustCompile("^[a-zA-Z0-9][a-zA-Z0-9_-]{2,32}$"),
+				"name must be between 3 and 33 characters long, start with an alphanumeric character, and contain only alphanumeric characters, dashes or underscores.",
+			),
+		},
+
+		"cognitive_account_id": commonschema.ResourceIDReferenceRequiredForceNew(&cognitiveservicesprojects.AccountId{}),
+
```

Comment:
I think it's missing the `display_name` and `description` attributes.

Reply:
`display_name` and `description` added, but due to API limitation we cannot update these properties to empty for now.